tests/phpunit/api/v3/JobTest.php

```diff
@@ -330,6 +330,307 @@ public function testBatchMergeWorks($dataSet) {
   }
 
   /**
+   * Test the decisions made for addresses when merging.
+   *
+   * @dataProvider getMergeAddresses
+   *
+   * Scenarios:
+   * (the ones with **** could be disputed as whether it is the best outcome).
+   *   'matching_primary' - Primary matches, including location_type_id. One contact has an additional address.
+   *     - result - primary is the shared one. Additional address is retained.
+   *   'matching_primary_reverse' - Primary matches, including location_type_id. Keep both. (opposite order)
```

Comment:
@JKingsnorth I wrote this test with the goal of focussing on some of the curlier scenarios - especially the hooks. That part is not complete - but it is, disturbingly, failing on the second check - it seems that it is failing to transfer the second address in this scenario....

        'matching_primary_reverse' => array(
           'contact_1' => array(
              array_merge(array('location_type_id' => 'Home', 'is_primary' => 1), $address1),
           ),
           'contact_2' => array(
               array_merge(array('location_type_id' => 'Home', 'is_primary' => 1), $address1),
               array_merge(array('location_type_id' => 'Work', 'is_primary' => 0), $address2),
           ),
           'expected' => array(
               array_merge(array('location_type_id' => 'Home', 'is_primary' => 1), $address1),
               array_merge(array('location_type_id' => 'Work', 'is_primary' => 0), $address2),
           ),